server/src/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -2627,7 +2627,7 @@ public boolean postStateTransitionEvent(final StateMachine2.Transition<State, Vi
                 s_logger.info("Schedule a router reboot task as router " + vo.getId() + " is powered-on out-of-band. we need to reboot to refresh network rules");
```

Comment:
@koushik-das The problem is probably not solved in the most elegant way. The bit (vo.getType() == VirtualMachine.Type.DomainRouter) could have been pulled up in an extra containing if block.
In the end the only the reprovision_out_of_band is making the final choice between alerting and rebooting